### ISSUE TITLE
`StackOutputs.stack_top()` fails when `stack <  STACK_TOP_SIZE`

### ISSUE THREAD

**Issue by frisitano** [0xPolygonMiden/miden-vm#1007]
When `StackOutputs.stack_top()` is called on a`StackOutput` instance that has `stack.len() < STACK_TOP_SIZE` it results in the following error:

```
 panicked at 'failed to convert vector to array: [BaseElement(6029861533644216910), BaseElement(14833700114938183983), BaseElement(14237865717728717124), BaseElement(8307047245961902131), BaseElement(0), BaseElement(0), BaseElement(0), BaseElement(0)]', /Users/f/dev/miden/miden-vm/core/src/stack/outputs.rs:83:14
```

This is caused by the `try_into()` failing because the stack vector is too short.  This is causing problems when trying to verify a miden transaction.

https://github.com/0xPolygonMiden/miden-vm/blob/3022d257eec9cdeca2273b0b66b633bcb3552274/core/src/stack/outputs.rs#L75-L84

We could add a `.resize(STACK_TOP_SIZE, ZERO)` to resize and pad the stack but I'm not sure if this is the desired behaviour.

**Comment by bobbinth** [comment]
This is a bit strange because stack output should always have at least 16 items (i.e., `STACK_TOP_SIZE`) in it. This is because we build stack outputs from the stack [here](https://github.com/0xPolygonMiden/miden-vm/blob/main/processor/src/stack/mod.rs) and the stack should always have at least 16 items.

So, the bug is either in the processor and for some reason stack depth drops to below 16 at some point, or we are initializing `StackOutputs` somewhere else - and that's causing the issue.

If it is the latter, we should do the following: in the constructors of `StackOutput` pad the input items with `ZERO` to ensure that `StackOutputs` always has at least 16 elements. One thing to pay attention to is the order of elements (I don't remember whether we'd need to pad from the front or from the back).

**Comment by frisitano** [comment]
> So, the bug is either in the processor and for some reason stack depth drops to below 16 at some point, or we are initializing `StackOutputs` somewhere else - and that's causing the issue.
> 
> If it is the latter, we should do the following: in the constructors of `StackOutput` pad the input items with `ZERO` to ensure that `StackOutputs` always has at least 16 elements. One thing to pay attention to is the order of elements (I don't remember whether we'd need to pad from the front or from the back).

Turns out the issue was related to construction of the `StackOutputs` struct. Padding appropriately and ordering fixed the issue.

**Comment by bobbinth** [comment]
Should we still update the `StackOutputs` constructor?

**Comment by frisitano** [comment]
> Should we still update the `StackOutputs` constructor?

Yes I think that makes sense. Removes a potential footgun.

**Comment by bobbinth** [comment]
Closed by #1010.
